client/client.go

```diff
@@ -97,7 +97,19 @@ func (ar *ApiRequest) Do(req *http.Request) (*http.Response, error) {
 	if req.Header.Get("Authorization") == "" {
 		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", ar.auth))
 	}
-	return ar.api.Do(req)
+	return logRequestIDWrapper(ar.api.Do(req))
+}
+
+// logRequestIDWrapper logs the request-id in case of an invalid response to the API
+func logRequestIDWrapper(r *http.Response, err error) (*http.Response, error) {
+	log.Info("data")
+	if err == nil {
+		rid := r.Header.Get("x-men-requestid")
+		if rid != "" && r.StatusCode != http.StatusOK {
+			log.Infof("requestid: %s ", rid)
```

Comment:
Should we also extend this error message, to add a bit more context here?

Reply:
yes, I was thinking the same. @maciejmrowiec, what do u want in the logs?

Reply:
we need context, do you guys use errors.Wrap in client code ?

Reply:
Yeh